[IMP] MRP: Resupply subcontracting update 19.0 (#18331)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/subcontracting/subcontracting_resupply.rst

Reviewer: ```suggestion

note:
This document uses the term 'company' to refer to the internal company requiring subcontracted materials, and the term 'subcontractor' to refer to the external vendor that handles the outsourced subcontracted materials. 
```

If you agree with my comment below- this is a good way to keep everyhting understood, and not use different terms. Please feel free to modify- I just think a nice clarifiction can really help.

Reviewer: I recommend linking "create a new one" to the doc or section of a doc that explains that

Reviewer: ```suggestion
On the product form, click the *Purchase* tab and add the product's subcontractor as a
```

Reviewer: ```suggestion
Configure product subcontractor
```
Because I want to be consistent with the terms nd not confuse anyone, i think this mightbe a better heading. And we can add "'also referred to as a 'vendor' somewhere in this seciton as well

Reviewer: ```suggestion
Configure the BoM
```

Reviewer: I'd add the link to the creating a BoM doc for the words "create a new one"

Reviewer: LOVE THIS!

Reviewer: ```suggestion
Finally, add all necessary components in the *Components* tab. To add a new component,
```

Reviewer: ```suggestion
click :guilabel:`Add a line`. Then, select the  :guilabel:`Component` using the drop-down menu
```

Reviewer: ```suggestion
, and specify the required :guilabel:`Quantity` in the corresponding field.
```
Just to cut down on repeated words

Reviewer: You'll have to rewrite this to be 2 steps each in their own line

Reviewer: Generally, I recommend not starting a section with an image, and have the image appear after the relevant text (so in this case, after the list of steps in the illustration)

Reviewer: ```suggestion
In the *Products* tab, click :guilabel:`Add a product` and
```

Reviewer: ```suggestion
select the subcontracted product. Next, enter the :guilabel:`Quantity` in the corresponding field.
```

Reviewer: add the truck icon in the validate button

Reviewer: This is just a general suggestion, but especially when theres lots to look at on a screen, I'll start my directions with where they go, then what they do.

So for this, I would say "Alternatively, open the Inventory app, and on the :guilabel:`Resupply Subcontractor` card, click the :guilabel:`(#) To Process` button.

The logic is, I was looking for where ot click- but if I knew the card name first, and the button second, it's faster/easier for me. Totally optional, but something I think about- since I know I find it helpful, so I assume there are others like me who would also think it's helpful.

.. |POs| replace:: :abbr:`PO (Purchase Orders)`
.. |BoM| replace:: :abbr:`BoM (Bill of Materials)`

> In resupply subcontracting, a contracting company (Your Company) supplies the components of its

Reviewer: ```suggestion
In resupply subcontracting, a company supplies the components of its
```
I just find it comfusing when I read it - i know what your aim is, but I think we can add exmaples or phrasing further down in the doc perhaps.

product to a subcontractor, who manufactures the product, then delivers the finished product to the

> contractor's warehouse.

Reviewer: ```suggestion
company's warehouse.
```

I feel like if we always refer to the database owners as "the company" instead of "your company", and the subcontractors as such, it might keep things more clear. So not go back and forth between 'contracting company', 'company', and 'contractor' - because I was getting confused =D

> This article covers how to configure a subcontracted product and walk through the resupply

Reviewer: ```suggestion
This document covers how to configure a subcontracted product and walk through the resupply
```
I know some people keep saying "articles" in the standup, but these are documents in the documentation project/website/etc, so we should always refer to them as such

> subcontracting process.

Reviewer: Option - you can link to the config and workflow sections in the corresponding words above - total extra, not needed!

=============

> To use the resupply subcontractor workflow, contractors must first configure products with a

Reviewer: ```suggestion
To use the resupply subcontractor workflow, companies must first configure products with a
```

I think it's very important to only use one specific term for both sides of the workflow - so you can chose whichever you think is best. Subcontractpoor is a given, since it's the term we use in the DB and it's generally known. The main person/company/etc is the only one you need to decide on. I do suggest company, since even in Runbot we have a "your ocmpan" default name - so I htink if we use "Company" and "Contractor" thorughout - ut oculd be a soliution. 

IF you agree with this- I suggest adding a note for the entire document, in the opening. I'll add the ocmment/suggestion - I just thought of that.

:ref:`vendor pricelist <purchase/products/pricelist>` and a subcontracting-type |BoM| specifying the
product's components.

The pricelist allows the contractor to purchase the product from the vendor (subcontractor) through
a |PO|. The |BoM| allows the product to be manufactured externally by the subcontractor and

> automatically configures the components to be resupplied by the contractor.

Reviewer: ```suggestion
The pricelist allows the company to purchase the product from the subcontractor through
a |PO|. The |BoM| allows the product to be manufactured externally by the subcontractor and
automatically configures the components to be resupplied by the company.
```

I have a personal dislike of calling the company the "contractor" because we use "subcontractor" and "contractor" a lot - and while it's not super confusing to most,  I find it a bit confusing. But like myother comment - if you prefer to call the origin company/DB owver the "contractpor
 just ensure every mention is ocnsistent, and the clarifying note in th ebeginning =)

vendor by clicking :guilabel:`Add a line`. Select the subcontractor in the :guilabel:`Vendor`

> drop-down menu.

Reviewer: ```suggestion
vendor by clicking :guilabel:`Add a line`. Select the subcontractor using the :guilabel:`Vendor`
drop-down menu.
```

Then, enter the price of the product in the :guilabel:`Unit Price` field.

> Finally, set a :doc:`lead time <../../inventory/warehouses_storage/replenishment/lead_times>` for

Reviewer: ```suggestion
Finally, set a :doc:`Lead Time <../../inventory/warehouses_storage/replenishment/lead_times>` for
```

> the product in the :guilabel:`Lead Time` field to specify the number of days for the subcontractor

Reviewer: ```suggestion
the product in the corresponding field to specify the number of days for the subcontractor
```
This way you don't repeat the same 2 words close to each other

to receive components, produce the product, and deliver the finished good.

.. note::
   Since contractors are not responsible for manufacturing the final product, there is no need to
   configure manufacturing lead times on a |BoM|. Instead, provide only a single *Lead Time* on the

>    vendor pricelist, factoring in the duration for the subcontractor to receive the components from

Reviewer: Because it's on the GUI, I'd either change "Lead Time" to a GUI to match in title casee, or all lowercase, as a term of art. Either way!

>    the contractor, manufacture the product, and deliver the finished good back to the contractor.

Reviewer: ```suggestion
   the contractor, manufacture the product, and deliver the finished goods back to the company.
```
Or whateve ryou decide to call it =)

> .. image:: subcontracting_resupply/resupply-subcontracting-bom.png

Reviewer: Just a tip - the max is actually 933 - so for wide images with lots of info, I highly recommend having th eimage be max width 933. (I checked this with Sam, our old tech master, and I have used that since).

> <manufacturing/subcontracting_resupply/workflow/create-po>` to purchase the product from the
> subcontractor (1).

Reviewer: ```suggestion
The resupply subcontracting workflow begins by: 

#. :ref:`Create a PO <manufacturing/subcontracting_resupply/workflow/create-po>` to purchase the product from the
subcontractor.
#.
```
I recommend changing this from a paragraph of instructions to a numbered list that corresponds with the steps. This will help the reader follow the steps easier - I know it will help me =D

I didn't want to add all the suggestions, but you get the idea:
1 -PO...
2- Confirm PO..
3..... etc etc etc

Author: good suggestion. I need to find a way to cleanly work this into a way that makes with the visual. this is something i'll try to do in a subsequent update across versions.

> The contractor (Your Company) then confirms the |PO|, which creates both a resupply order to

Reviewer: ```suggestion
#. The company confirms the |PO|, which creates both a resupply order to
```

transfer the components and a receipt to receive the final product (2) from the subcontractor.

> Next, the contractor :ref:`validates the transfer

Reviewer: ```suggestion
#. The company :ref:`validates the transfer
```

> :guilabel:`Quantity` field.

Reviewer: ```suggestion
```

After adding the product, the :guilabel:`Expected Arrival` field is updated with the finished
product's expected delivery date, as configured earlier with the vendor *Lead Time*.

Finally, click :guilabel:`Confirm Order` to confirm the |PO|. A receipt and a resupply order are

> automatically created, accessible via the :guilabel:`Receipt` and :guilabel:`Resupply` smart buttons

Reviewer: I'd add the icons in the buttons as well (the truck for both).

---------------

After the resupply order is confirmed, the subcontractor manufactures the product and delivers the

> finished good back to the contractor.

Reviewer: ```suggestion
finished good back to the company.
```

> To receive the finished product from the subcontractor, click the :guilabel:`Receive Products`

Reviewer: The button I see only sais "Receive" but I might have set it up wrong

> button on the |PO|, or click the :guilabel:`Receipt` smart button at the top of the page. Then,

Reviewer: You may want to expand on this, since we are still on the resupply form. I have a "Source PO" button I used to get to the PO.

Alternatively, you can treat every section lijke  a new part, and here you can start with 'to receive the finishe dproduct, open theorigina PO by...." and you cna list the 2 ways ot get to it.

---------------------

After validating the receipt, Odoo automatically generates inventory moves to track the movement of
subcontracted products between :doc:`locations
<../../inventory/warehouses_storage/inventory_management/use_locations>`. To view these inventory
moves, navigate to :menuselection:`Inventory app --> Reporting --> Moves History`.

In resupply subcontracting, Odoo first transfers any product components to a dedicated location
called *Subcontracting*. Another location called *Production* then consumes the components and
produces the finished good. Once produced, the good then moves back to the *Subcontracting* location

> before finally entering the contractor's stock when the receipt is validated.

Reviewer: ```suggestion
before finally entering the company's stock when the receipt is validated.
```
